pilot/pkg/xds/delta.go

```diff
@@ -635,15 +638,29 @@ func deltaToSotwRequest(request *discovery.DeltaDiscoveryRequest) *discovery.Dis
 }
 
 // deltaWatchedResources returns current watched resources of delta xds
-func deltaWatchedResources(existing []string, request *discovery.DeltaDiscoveryRequest) ([]string, bool) {
```

Comment:
Is this the main perf bottle neck for ambient? I would prefer optimizing this function instead of changing the watched resources `struct`.

Reply:
So this overall PR makes ambient about 100x more efficient. About 25% comes from the changes to this function and 75% come from the workload.go changes. 

The change here impacts all xds types positively - not just ambient.

Reply:
That's the risk,  map is sometimes less mem efficient than slice

Reply:
https://github.com/planetscale/vtprotobuf this pkg is actully less active, maybe should ask earlier, what are the companies behind that maintains the project, will it continue evolve with protobuf

Reply:
No, I have not compared sidecar performance. 

While a map may use more memory, it is far less efficient to constantly translate between slice and map back and forth like we do today. Just look at the graph in the PR description.. before we were allocating 300mb/s and now it's nearly nothing. 

I am a bit confused what the concern is .. from my perspective this PR is one of the biggest performance improvements we have ever made